sgkit/tests/test_conversion.py

```diff
@@ -0,0 +1,68 @@
+from typing import Any
+
+import dask.array as da
+import numpy as np
+import pytest
+from xarray import Dataset
+
+from sgkit import variables
+from sgkit.stats.conversion import convert_probability_to_call
+from sgkit.testing import simulate_genotype_call_dataset
+
+
+def simulate_dataset(gp: Any, chunks: int = -1) -> Dataset:
+    gp = da.asarray(gp)
+    gp = gp.rechunk((None, None, chunks))
+    ds = simulate_genotype_call_dataset(n_variant=gp.shape[0], n_sample=gp.shape[1])
+    ds = ds.drop_vars([variables.call_genotype, variables.call_genotype_mask])
+    ds = ds.assign(
+        {
+            variables.call_genotype_probability: (
+                ("variants", "samples", "genotypes"),
+                gp,
+            )
+        }
+    )
+    return ds
+
+
+@pytest.mark.parametrize("chunks", [-1, 1])
+@pytest.mark.parametrize("dtype", ["f4", "f8"])
+def test_convert_probability_to_call(chunks: int, dtype: str) -> None:
+    ds = simulate_dataset(
+        [
+            # hom/ref, het, and hom/alt respectively
+            [[0.98, 0.01, 0.01], [0.01, 0.98, 0.01], [0.01, 0.01, 0.98]],
+            # Missing combos (should all be no call)
+            [[0.5, 0.5, np.nan], [1.0, np.nan, np.nan], [np.nan, np.nan, np.nan]],
+            # No probability above threshold and no max probability
+            [[0.4, 0.4, 0.2], [0.0, 1.0, 1.0], [0.3, 0.3, 0.3]],
+        ],
+        chunks,
+    )
+    ds[variables.call_genotype_probability] = ds[  # type: ignore[no-untyped-call]
+        variables.call_genotype_probability
+    ].astype(dtype)
+    ds = convert_probability_to_call(ds)
```

Comment:
A test with a different threshold (and 0) would be good.

Reply:
https://github.com/pystatgen/sgkit/pull/419/commits/61176be7f6d943ee12d28057086f73c13d3fdb5b#diff-9d395b63a7f182b2d203eb33ff05b763f3150a23a27050d8896d2a2a25b1ba5cR60-R86